src/pubsub/ua_pubsub_reader.c

```diff
@@ -79,21 +311,30 @@ UA_Server_removeReaderGroup(UA_Server *server, UA_NodeId groupIdentifier) {
         return UA_STATUSCODE_BADNOTFOUND;
     }
 
+    if(readerGroup->configurationFrozen){
+        UA_LOG_WARNING(&server->config.logger, UA_LOGCATEGORY_SERVER,
+                       "Remove ReaderGroup failed. ReaderGroup is frozen.");
+        return UA_STATUSCODE_BADCONFIGURATIONERROR;
+    }
+
     /* Search the connection to which the given readergroup is connected to */
     UA_PubSubConnection *connection =
         UA_PubSubConnection_findConnectionbyId(server, readerGroup->linkedConnection);
     if(connection == NULL) {
         return UA_STATUSCODE_BADNOTFOUND;
     }
 
-    /* Unregister subscribe callback */
-    UA_PubSubManager_removeRepeatedPubSubCallback(server, readerGroup->subscribeCallbackId);
+    if(readerGroup->state == UA_PUBSUBSTATE_OPERATIONAL) {
+        /* Unregister subscribe callback */
+        UA_PubSubManager_removeRepeatedPubSubCallback(server, readerGroup->subscribeCallbackId);
```

Comment:
I need some logical clarification: Why is the repeated callback removed if the state is OPERATIONAL?

Reply:
The removeReaderGroup will be called while server cleanup and the callback will be only active if the state is operational. If we do not have this check we may result in double free as there won't be any callbacks.